src/expr/src/expr/expr_udf.rs

```diff
@@ -79,19 +73,20 @@ impl Expression for UdfExpression {
 impl UdfExpression {
     async fn eval_inner(
         &self,
-        columns: Vec<arrow_array::ArrayRef>,
+        columns: Vec<ArrayRef>,
         vis: &risingwave_common::buffer::Bitmap,
     ) -> Result<ArrayRef> {
-        let opts = arrow_array::RecordBatchOptions::default().with_row_count(Some(vis.len()));
-        let input =
-            arrow_array::RecordBatch::try_new_with_options(self.arg_schema.clone(), columns, &opts)
-                .expect("failed to build record batch");
+        let chunk = DataChunk::new(columns, vis.clone());
+        let visible_rows: Vec<usize> = vis.iter_ones().collect();
+        let compacted_chunk = chunk.compact_cow();
+        let input = arrow_array::RecordBatch::try_from(compacted_chunk.as_ref())?;
```

Comment:
We should keep using `try_new_with_options` here. Because the input may contain no column, in which case `try_from` will panic. The UDTF side should also be fixed.